vslib/src/SwitchStateBase.cpp

```diff
@@ -2662,6 +2668,111 @@ sai_status_t SwitchStateBase::refresh_system_port_list(
     return SAI_STATUS_SUCCESS;
 }
 
+sai_status_t SwitchStateBase::createVoqSystemNeighborEntry(
+        _In_ const std::string &serializedObjectId,
+        _In_ sai_object_id_t switch_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list)
+{
+    SWSS_LOG_ENTER();
+
+    // For VOQ switch, encap index attribute should be set for local neighbors. For
+    // remote neighbors check for presence of encap index supplied from upper layer.
+    // And also validate isLocal and impose encap index flags in the attribute list
+
+    // Determine whether this neighbor is local neighbor or remote neighbor by checking the
+    // RIF id provided. If the port of the RIF is system port, the neighbor is remote neighbor
+
+    bool is_system_neigh = false;
+    sai_attribute_t attr;
+    sai_neighbor_entry_t nbr_entry;
+
+    sai_deserialize_neighbor_entry(serializedObjectId, nbr_entry);
+
+    attr.id = SAI_ROUTER_INTERFACE_ATTR_PORT_ID;
+
+    CHECK_STATUS(get(SAI_OBJECT_TYPE_ROUTER_INTERFACE, nbr_entry.rif_id, 1, &attr));
+
+    if (objectTypeQuery(attr.value.oid) == SAI_OBJECT_TYPE_SYSTEM_PORT)
+    {
+        is_system_neigh = true;
+    }
+
+    uint32_t encap_index = 0;
+    bool impose_encap_index = false;
+    bool is_local = true;
+
+    for (uint32_t i = 0; i < attr_count; i++)
+    {
+        switch (attr_list[i].id)
+        {
+            case SAI_NEIGHBOR_ENTRY_ATTR_ENCAP_INDEX:
+
+                encap_index = attr_list[i].value.u32;
+
+                break;
+
+            case SAI_NEIGHBOR_ENTRY_ATTR_ENCAP_IMPOSE_INDEX:
+
+                impose_encap_index = attr_list[i].value.booldata;
+
+                break;
+
+            case SAI_NEIGHBOR_ENTRY_ATTR_IS_LOCAL:
+
+                is_local = attr_list[i].value.booldata;
+
+                break;
+
+            default:
+                break;
+        }
+    }
+
+    if (impose_encap_index && encap_index == 0)
+    {
+        SWSS_LOG_ERROR("Impose invalid encap index %d for %s neigh %s",
+                encap_index,
+                (is_system_neigh ? "VOQ" : "local"),
+                serializedObjectId.c_str());
+
+        return SAI_STATUS_FAILURE;
+    }
+
+    if (is_system_neigh && is_local)
+    {
+        SWSS_LOG_ERROR("VOQ neigh info mismatch for %s. RIF is remote. attr is_local is true", serializedObjectId.c_str());
+
+        return SAI_STATUS_FAILURE;
+    }
+
+    if(!is_local && encap_index == 0)
```

Comment:
space after if, and remove empty line 2751

Reply:
Thanks. I'll fix them

Reply:
Fixed